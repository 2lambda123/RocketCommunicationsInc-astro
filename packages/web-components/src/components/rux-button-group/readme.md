# Grouped Buttons

Common button groupings follow these conventions:

-   Cancel buttons are always presented to the left of actions such as “Submit.”

-   Buttons within the same group should maintain their inherent size. Do not stretch one button to match another’s width.

## Web Component Usage

### 1. Render the Astro Button Group Web Component

```xml
<rux-button-group>
  <rux-button>Button 1</rux-button>
  <rux-button>Button 2</rux-button>
</rux-button-group>
```

By default button group aligns buttons to the left. Alternatively an `h-align` property may be passed to change alignment to `"center"` or `"right"`.

```xml
<rux-button-group h-align="right">
  <rux-button>Button 1</rux-button>
  <rux-button>Button 2</rux-button>
</rux-button-group>
```

<!-- Auto Generated Below -->


> **[DEPRECATED]** Button Group is deprecated and will be removed in a next major release.
Instead, you should use `flex` or `grid` in combination with our spacing design tokens.

## Properties

| Property | Attribute | Description                                          | Type                            | Default  |
| -------- | --------- | ---------------------------------------------------- | ------------------------------- | -------- |
| `hAlign` | `h-align` | The horizontal alignment of buttons within the group | `"center" \| "left" \| "right"` | `'left'` |


## Shadow Parts

| Part          | Description |
| ------------- | ----------- |
| `"container"` |             |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
