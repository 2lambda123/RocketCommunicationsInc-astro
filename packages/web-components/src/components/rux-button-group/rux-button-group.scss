:host {
    display: inline;
}

:host([hidden]) {
    display: none;
}

.rux-button-group {
    display: flex;
    ::slotted(rux-button:not(:last-child)) {
        margin-right: var(--spacing-2);
    }

    &--left {
        justify-content: flex-start;
    }

    &--right {
        justify-content: flex-end;
    }

    &--center {
        justify-content: center;
    }
}
