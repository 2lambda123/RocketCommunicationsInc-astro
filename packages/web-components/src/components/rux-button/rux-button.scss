:host {
    /**
     * @prop --button-text-color: [DEPRECATED] Button text color
     */
    --button-text-color: var(--color-text-inverse);

    /**
     * @prop --button-background-color: [DEPRECATED] Button background color
     */
    --button-background-color: var(--color-background-interactive-default);

    /**
     * @prop --button-border-color: [DEPRECATED] Button border color
     */
    --button-border-color: transparent;

    /**
     * @prop --button-hover-background-color: [DEPRECATED] Button hover background color
     */
    --button-hover-background-color: var(--color-background-interactive-hover);

    /**
     * @prop --button-hover-border-color: [DEPRECATED] Button hover border color
     */
    --button-hover-border-color: transparent;

    /**
     * @prop --button-active-background-color: [DEPRECATED] Button active background color
     */
    --button-active-background-color: var(
        --color-background-interactive-default
    );

    /**
     * @prop --button-active-border-color: [DEPRECATED] Button active border color
     */
    --button-active-border-color: var(--color-background-interactive-default);

    /**
     * @prop --button-secondary-text-color: [DEPRECATED] Button secondary text color
     */

    --button-secondary-text-color: var(--color-background-interactive-default);

    /**
    * @prop --button-secondary-hover-text-color: [DEPRECATED] Secondary button hover text color
    */
    --button-secondary-hover-text-color: var(
        --color-background-interactive-hover
    );

    /**
     * @prop --button-secondary-background-color: [DEPRECATED] Button secondary background color
     */
    --button-secondary-background-color: transparent;

    /**
     * @prop --button-secondary-border-color: [DEPRECATED] Button secondary border color
     */
    --button-secondary-border-color: var(
        --color-background-interactive-default,
        rgb(0, 90, 143)
    );

    /**
     * @prop --button-secondary-hover-background-color: [DEPRECATED]  Button secondary hover background color
     */
    --button-secondary-hover-background-color: transparent;

    /**
     * @prop --button-secondary-hover-border-color: [DEPRECATED]  Button secondary hover border color
     */
    --button-secondary-hover-border-color: var(
        --color-background-interactive-hover
    );

    /**
    * @prop --button-borderless-text-color: [DEPRECATED] Button borderless text color
    */
    --button-borderless-color: var(--color-background-interactive-default);

    /**
    * @prop --button-borderless-hover-color: [DEPRECATED] Borderless buttons hover text color 
    */
    --button-borderless-hover-color: var(--color-background-interactive-hover);

    display: inline-block;
    width: auto;
}

:host([hidden]) {
    display: none;
}

:host([disabled]) {
    pointer-events: none;
}
.rux-button {
    display: flex;
    position: relative;
    margin: 0;
    width: inherit;
    padding: 0 1rem;
    height: 2.125rem;
    min-width: 2.25rem;
    border-radius: var(--radius-base);
    color: var(--button-text-color);
    font-family: var(--font-body-1-font-family);
    font-size: var(--font-body-1-font-size);
    font-weight: var(--font-body-1-font-weight);
    letter-spacing: var(--font-body-1-letter-spacing);
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    justify-content: center;
    align-items: center;
    user-select: none;

    rux-icon,
    ::slotted(rux-icon) {
        height: 1rem;
        width: 1rem;
        margin-right: 0.625rem;
        color: var(--button-text-color);
    }

    &:disabled {
        opacity: var(--disabled-opacity);
        cursor: not-allowed;
        pointer-events: none;
    }

    &:focus {
        outline: none;
    }

    &:hover:not([disabled]) {
        cursor: pointer;
    }

    &--default {
        border: 1px solid var(--button-border-color);
        background-color: var(--button-background-color);
        &:hover:not([disabled]) {
            border-color: var(--button-hover-border-color);
            background-color: var(--button-hover-background-color);
        }
        &:active:not([disabled]) {
            border-color: var(--button-active-border-color);
            background-color: var(--button-active-background-color);
        }
    }

    &--secondary {
        color: var(--button-secondary-text-color);
        background-color: var(--button-secondary-background-color);
        border: 1px solid var(--button-secondary-border-color);

        &:hover:not([disabled]) {
            color: var(--button-secondary-hover-text-color);
            background-color: var(--button-secondary-hover-background-color);
            border-color: var(--button-secondary-hover-border-color);
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-secondary-hover-text-color);
            }
        }

        &:active:not([disabled]) {
            border-color: var(--button-secondary-border-color);
            background-color: var(--button-secondary-background-color);
            color: var(--button-secondary-text-color);
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-secondary-text-color);
            }
        }
        rux-icon,
        ::slotted(rux-icon) {
            color: var(--button-secondary-text-color);
        }
    }
    &--borderless {
        color: var(--button-borderless-color);
        border: none;
        background: none;
        &:hover:not([disabled]) {
            color: var(--button-borderless-hover-color);
            background: none;
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-borderless-hover-color);
            }
        }
        &:active:not([disabled]) {
            color: var(--button-borderless-color);
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-borderless-color);
            }
        }
        rux-icon,
        ::slotted(rux-icon) {
            color: var(--button-borderless-color);
        }
    }

    &--small {
        height: 1.625rem;
        padding: 0 1rem;
        line-height: 1;
    }

    &--large {
        height: 2.875rem;
        min-width: 3rem;
        padding: 0 1rem;
    }

    &--icon-only {
        font-size: 0;
        width: 3rem;
    }
}

.rux-button--icon-only ::slotted(rux-icon),
.rux-button--icon-only rux-icon {
    margin-left: -0.625rem;
    margin-right: -0.625rem;
}
