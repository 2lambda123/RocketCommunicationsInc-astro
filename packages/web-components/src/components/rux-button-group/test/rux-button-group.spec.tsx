import { newSpecPage } from '@stencil/core/testing'
import { RuxButtonGroup } from '../../rux-button-group/rux-button-group'
import { RuxButton } from '../../rux-button/rux-button'

describe('rux-button-group', () => {
    it('builds', async () => {
        const buttonGroup = new RuxButtonGroup()

        expect(buttonGroup).toBeTruthy()
        expect(buttonGroup).toEqual({
            hAlign: 'left',
        })
    })

    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxButtonGroup, RuxButton],
            html: `<rux-button-group>
              <rux-button icon="" secondary>Button</rux-button>
              <rux-button icon="">Button</rux-button>
            </rux-button-group>`,
        })
        expect(page.root).toMatchSnapshot()
    })

    it('aligns', async () => {
        const page = await newSpecPage({
            components: [RuxButtonGroup],
            html: `<rux-button-group h-align="right"></rux-button-group>`,
        })

        expect(page.root).toMatchSnapshot()
    })
})
