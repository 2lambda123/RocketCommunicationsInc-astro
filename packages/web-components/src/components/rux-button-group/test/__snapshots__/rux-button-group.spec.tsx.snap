// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-button-group aligns 1`] = `
<rux-button-group h-align="right">
  <mock:shadow-root>
    <div class="rux-button-group rux-button-group--right">
      <slot></slot>
    </div>
  </mock:shadow-root>
</rux-button-group>
`;

exports[`rux-button-group renders 1`] = `
<rux-button-group>
  <mock:shadow-root>
    <div class="rux-button-group rux-button-group--left">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <rux-button icon="" secondary="">
    <mock:shadow-root>
      <button class="rux-button rux-button--secondary" type="button">
        <slot></slot>
      </button>
    </mock:shadow-root>
    Button
  </rux-button>
  <rux-button icon="">
    <mock:shadow-root>
      <button class="rux-button rux-button--default" type="button">
        <slot></slot>
      </button>
    </mock:shadow-root>
    Button
  </rux-button>
</rux-button-group>
`;
